web/src/echo.js

```diff
@@ -445,6 +445,12 @@ export function process_from_server(messages) {
         // doing it unconditionally.
         for (const msg_list of message_lists.all_rendered_message_lists()) {
             msg_list.view.rerender_messages(msgs_to_rerender);
+            // If the rerendering of the messages doesn't cause the current
+            // message list view to scroll, then we want to update the
+            // rerendered message header that may have been sticky.
+            if (msg_list === message_lists.current && narrow_state.is_message_feed_visible()) {
+                message_lists.current.view.update_sticky_recipient_headers();
```

Comment:
Ideally, `message_lists.current.view.update_sticky_recipient_headers();` call should be placed inside `_rerender_header`. But since `_rerender_header` is only being used inside `rerender_messages` in a for loop. We can add insert `message_lists.current.view.update_sticky_recipient_headers()` after the loop with a comment warning inside `_rerender_header` about this. Something like "this method should only be called with `rerender_messages`" or we assume that it does.